pymc/distributions/continuous.py

```diff
@@ -567,14 +568,20 @@ def rng_fn(
         upper: Union[np.ndarray, float],
         size: Optional[Union[List[int], int]],
     ) -> np.ndarray:
+        # Upcast to float64 and then downcast if necessary
+        #   (Work-around for https://github.com/scipy/scipy/issues/15928)
+        if cls.dtype == "floatX":
+            dtype = config.floatX
+        else:
```

Comment:
Hmm doesn't the perform method which calls this downcast already based on self.dtype? In that case we don't need this "hack" here. Specially because config.floatX could change after the RV is defined

Reply:
I'm not sure I follow :confused: . I just see that `rng_fn` is a classmethod here. It will only have the `cls` attribute (?)

Reply:
This classmethod is always called by the baseclass perform method which already casts to the right dtype if needed: https://github.com/pymc-devs/pytensor/blob/7ecb9f8c6b6a2eac940947bac955a10785240667/pytensor/tensor/random/op.py#L342

So we shouldn't have to worry about it here, just upcast the inputs to float64

Reply:
I see! I've removed the downcasting here https://github.com/pymc-devs/pymc/pull/7026/commits/93f9f3bd0e3320ab2162ec8b4ae833d931d58f69